app/models/ability.rb

```diff
@@ -5,6 +5,7 @@ def initialize(user)
     if user
       can %i(create read destroy update), Wishlist, user_id: user.id
       can %i(create update), Item, wishlist: { user_id: user.id }
+      can %i(destroy), Reservation
```

Comment:
With such rule, a user can destroy any other Reservation he wishes, even the ones not on his items.